app/javascript/application.js

```diff
@@ -17,28 +17,50 @@ if ($inputs) {
   }
 }
 
-//enhance all tagged select elements to be accessible-autocomplete elements
-const $acElements = document.querySelectorAll('[data-module="accessible-autocomplete"]')
-if ($acElements) {
-  for (let i = 0; i < $acElements.length; i++) {
-    const name = $acElements[i].getAttribute('data-name')
-
-    accessibleAutocomplete.enhanceSelectElement({
-      selectElement: $acElements[i],
-      defaultValue: '',
-      showNoOptionsFound: name === null,
-      name: name
-    })
-  }
-}
-
 // Avoid flickering header menu on small screens
 // Refer to `stylesheets/local/custom.scss`
 const $headerNavigation = document.querySelector('ul.app-header-menu-hidden-on-load')
 if ($headerNavigation) {
   $headerNavigation.classList.remove("app-header-menu-hidden-on-load")
 }
 
+convertSelectToAutocomplete()
+
 $(document).on('turbo:render', function () {
   initAll()
-})
\ No newline at end of file
+  convertSelectToAutocomplete()
+})
+
+function convertSelectToAutocomplete(){
+  //enhance all tagged select elements to be accessible-autocomplete elements
+  const $acElements = document.querySelectorAll('[data-module="accessible-autocomplete"]')
+  if ($acElements) {
+    for (let i = 0; i < $acElements.length; i++) {
+      var convertedToAutocomplete = $acElements[i].getAttribute('data-converted')
+      if(!convertedToAutocomplete){
+        const name = $acElements[i].getAttribute('data-name')
+        const autoselect = trueFalseToBoolean($acElements[i].getAttribute('data-autoselect'))
```

Comment:
Adding a 10-line method twice feels a bit overkill when you could just do  `$acElements[i].getAttribute('data-autoselect') === true` here

Reply:
rails stores all data attributes as strings - but yeah I think the method I did is very inelegant, I'll do your suggestion (with the minor change that the conditional statement compares against the string "true" instead of a boolean)